devicetypes/sky-nie/in-wall-smart-switch.src/in-wall-smart-switch.groovy

```diff
@@ -0,0 +1,362 @@
+/**
+ *  In-Wall Smart Switch v1.0.0
+ *
+ *  	Models: MS10ZS/MS12ZS/ZW30/ZW30S/ZW30TS
+ *
+ *  Author:
+ *   winnie (sky-nie)
+ *
+ *	Documentation:
+ *
+ *  Changelog:
+ *
+ *    1.0.0 (12/22/2021)
+ *      - Initial Release
+ *
+ * Reference：
+ *  https://github.com/krlaframboise/SmartThings/blob/master/devicetypes/krlaframboise/eva-logik-in-wall-smart-switch.src/eva-logik-in-wall-smart-switch.groovy
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+import groovy.json.JsonOutput
+
+metadata {
+    definition (name: "In-Wall Smart Switch", namespace: "sky-nie", author: "winnie", mnmn: "SmartThings", vid:"generic-switch") {
```

Comment:
@sky-nie Did You try to fix "Min Smart Plug" in a first place instead of duplicating most of the DTH code?